packages/eslint-plugin/src/rules/no-type-imports-with-verbatim-side-effects.ts

```diff
@@ -0,0 +1,78 @@
+import type { TSESLint, TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+import * as util from '../util';
+
+type Options = [];
+type MessageIds = 'useTopLevelQualifier';
+
+export default util.createRule<Options, MessageIds>({
+  name: 'no-type-imports-with-verbatim-side-effects',
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'Enforce the use of top-level import type qualifier when an import only has specifiers with inline type qualifiers',
+      recommended: false,
+    },
+    fixable: 'code',
+    messages: {
+      useTopLevelQualifier:
+        'TypeScript will only remove the inline type specifiers which will leave behind a side effect import at runtime. Convert this to a top-level type qualifier to properly remove the entire import.',
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    const sourceCode = context.getSourceCode();
+    return {
+      ImportDeclaration(node): void {
+        if (node.importKind === 'type') {
+          return;
+        }
```

Comment:
[Nit] If you don't mind the more complicated query? Not blocking at all.

```suggestion
      'ImportDeclaration[importKind!="type"]'(
        node: TSESTree.ImportDeclaration,
      ): void {
```

Reply:
You could probably fandangle something by chaining `:has` and `:not` or something - but yeah I don't think it's super worth it.

In the beginning I used to love loading a bunch of stuff into the selectors. Now a days I think there's definitely a trade-off with selectors because it's a DSL and they're not the simplest thing to understand.